test/SerialIOTest.cpp

```diff
@@ -1642,15 +1642,37 @@ void fileBased_write_test(const std::string & backend)
         o.iterations[ 6 ]
             .particles[ "e" ][ "position" ][ "x" ]
             .makeConstant< double >( 1.0 );
+
+        // additional iteration with over-running iteration padding but similar content
+        //                  padding:    000
+        uint64_t const overlong_it = 123456;
+        o.iterations[ overlong_it ];
+        // write something to trigger opening of the file
+        o.iterations[ overlong_it ].particles[ "e" ][ "position" ][ "x" ].resetDataset(
+                { Datatype::DOUBLE, { 12 } } );
+        o.iterations[ overlong_it ]
+                .particles[ "e" ][ "position" ][ "x" ]
+                .makeConstant< double >( 1.0 );
+
+        o.iterations[ overlong_it ].setTime(static_cast< double >(overlong_it));
+        REQUIRE(o.iterations.size() == 7);
     }
-    REQUIRE((auxiliary::file_exists("../samples/subdir/serial_fileBased_write00000004." + backend)
-        || auxiliary::directory_exists("../samples/subdir/serial_fileBased_write00000004." + backend)));
+    REQUIRE((auxiliary::file_exists("../samples/subdir/serial_fileBased_write004." + backend)
+        || auxiliary::directory_exists("../samples/subdir/serial_fileBased_write004." + backend)));
+    REQUIRE((auxiliary::file_exists("../samples/subdir/serial_fileBased_write123456." + backend)
+        || auxiliary::directory_exists("../samples/subdir/serial_fileBased_write123456." + backend)));
 
-    // additional iteration with different iteration padding but similar content
+    // additional iteration with shorter iteration padding but similar content
     {
         Series o = Series("../samples/subdir/serial_fileBased_write%01T." + backend, Access::READ_WRITE);
 
-        REQUIRE(o.iterations.empty());
+        REQUIRE(o.iterations.size() == 1);
+        /*
+         * 123456 has no padding, it's just a very long number.
+         * So even when opening the series with a padding of 1,
+         * that iteration will be opened.
+         */
+        REQUIRE(o.iterations.count(123456) == 1);
```

Comment:
I was initially thinking here: should we read `123456` if our padding was specifically asked to be `00`?

But if we would skip it, then this would be a bit inconsistent with reading unpadded numbers.
Thus, I think the logic change here in the test that you added is the proper way to handle this :+1:

Reply:
> should we read 123456 if our padding was specifically asked to be 00?

Yep, it should, because that number is not padded ;)